### PR TITLE
[Merged by Bors] - TY-2208 update key phrases [2]

### PR DESCRIPTION
**Pull request by janpetschexain**
**References**

- [TY-2208]
- #348
- #361
- #371

**Summary**

- make `SMBert` available to `CoiSystem` & replace the smbert closure
- mock `SMBert` in the coi tests & adjust reranker builder
- add more key phrase selection tests

[TY-2208]: https://xainag.atlassian.net/browse/TY-2208?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: xayn-ai/src/coi/mod.rs [hunk 1]
```diff
@@ -6,6 +6,7 @@ mod stats;
 mod system;
 mod utils;
 
+#[cfg(not(test))]
```
Comment (acrrd) [inline xayn-ai/src/coi/mod.rs:9]: Why we need this?

File: xayn-ai/Cargo.toml [hunk 2]
```diff
@@ -8,11 +8,12 @@ edition = "2018"
 anyhow = "1.0.52"
 bincode = "1.3.3"
 derivative = "2.2.0"
-derive_more = { version = "0.99.17", default-features = false, features = ["from", "display"] }
+derive_more = { version = "0.99.17", default-features = false, features = ["deref", "display", "from"] }
 displaydoc = "0.2.3"
 itertools = "0.10.3"
 layer = { path = "../layer" }
 lazy_static = "1.4.0"
+mockall_double = "0.2.1"
```
Comment (acrrd) [inline xayn-ai/Cargo.toml:16]: I tried this in the past but it was impossible to have test with real implementation and mock during tests.

File: xayn-ai/src/reranker/public.rs [hunk 3]
```diff
@@ -208,21 +212,25 @@ impl<SV, SM, QAV, QAM, DM> Builder<SV, SM, QAV, QAM, DM> {
...
             .with_token_size(90)?
             .with_accents(false)
             .with_lowercase(true)
             .with_pooling(AveragePooler)
             .build()?;
-        let coi = CoiSystemImpl::new(CoiSystemConfiguration::default());
+        #[cfg(not(test))]
+        let coi = CoiSystemImpl::new(CoiSystemConfiguration::default(), smbert.clone());
+        #[cfg(test)]
+        let coi = CoiSystemImpl::default();
```
Comment (acrrd) [inline xayn-ai/src/reranker/public.rs:233]: The smbert should be the same also during tests.

File: xayn-ai/src/coi/mod.rs (hunk 1 above)
Comment (janpetschexain) [reply xayn-ai/src/coi/mod.rs:9]: because clippy complains otherwise

**Review by acrrd — APPROVED** [review]

**Comment by janpetschexain** [comment]
bors merge

**Comment by janpetschexain** [comment]
bors merge

**Comment by janpetschexain** [comment]
bors merge

**Pull request closed without merging**